### ISSUE TITLE
Incorrect initilization of GPA crashes compiler with segfault

### ISSUE THREAD

**Issue by zxubian** [ziglang/zig#14190]
### Zig Version

0.11.0-dev.1161+a44085dc2

### Steps to Reproduce and Observed Behavior

``` zig
// repro.zig
const std = @import("std");

const allocator = std.heap.GeneralPurposeAllocator(.{}).allocator({});

pub fn main() !void {
    _ = allocator;
}
```
` zig run repro.zig`

### Observed Behaviour:
> Segmentation fault

### Expected Behavior

A helpful compiler error is displayed. For reference, the correct way to initialize the general purpose allocator is:

```zig
  var gpa = std.heap.GeneralPurposeAllocator(.{}){};
  const allocator = gpa.allocator();
```

**Comment by matu3ba** [comment]
It would be of help, if you could suggest what compiler error you would think be helpful.

**Comment by Vexu** [comment]
Duplicate of #12765, fixed by #13017
